status/discover.go

```diff
@@ -42,14 +36,11 @@ func DiscoverGrafana() (url string, err error) {
 	conf := config.Get()
 	// If display link is disable in Grafana configuration return empty string and avoid discover
 	if !conf.ExternalServices.Grafana.DisplayLink {
-		log.Debugf("Grafana display link is disabled in the configuration.")
 		return "", nil
 	}
 	if conf.ExternalServices.Grafana.URL != "" {
-		log.Debugf("Detected url %s configured for tracing service", config.Get().ExternalServices.Grafana.URL)
 		return conf.ExternalServices.Grafana.URL, nil
 	}
-	log.Debugf("Kiali is looking for Grafana service ...")
 	url, err = discoverUrlService(config.Get().ExternalServices.Grafana.ServiceNamespace, config.Get().ExternalServices.Grafana.Service)
```

Comment:
I see you took out all the debug messages, but you should have AT LEAST one debug message that logs what the newly discovered URL is. This is going to be helpful to let us know that this discovery actually was performed and it let's us know what it found.

Same is true for both the grafana and jaeger discovery... log a message for each one that is found.